Introduce "--vt-short-names-when-config" to switch test names display

PaulYuuu opened the pull request:
Sometimes, when we use the config file to run avocado, we want to use
short names as test names, so "--vt-short-names-when-config" is
introduced to allow users to switch it.

ID: 1905350 
Signed-off-by: Yihuang Yu <yihyu@redhat.com>

PaulYuuu commented:
@luckyh @yanan-fu @XueqiangWei, revert commit c41cd560ceccec71c7678741d6713ba05de0bb58  to adapt to our needs.

PaulYuuu commented:
Hello @clebergnu @pevogam, In some scenarios(vt.config), we need to make shortname equal to testname, so I reverted that commit and added a comment. If this bothers you, please let me know, thanks.

clebergnu commented:
> Hello @clebergnu @pevogam, In some scenarios(vt.config), we need to make shortname equal to testname, so I reverted that commit and added a comment. If this bothers you, please let me know, thanks.

Hi @PaulYuuu , I clearly did not understand this use case when I proposed the original change.  Now, I understand that there is a valid use case, but I wonder if that shouldn't be an opt-in type of behavior.

So, what about we enable that feature if an Avocado-level configuration such as `vt.short_names_when_config` is set to `True` (having a default of `False`).

PaulYuuu commented:
> > Hello @clebergnu @pevogam, In some scenarios(vt.config), we need to make shortname equal to testname, so I reverted that commit and added a comment. If this bothers you, please let me know, thanks.
> 
> Hi @PaulYuuu , I clearly did not understand this use case when I proposed the original change. Now, I understand that there is a valid use case, but I wonder if that shouldn't be an opt-in type of behavior.
> 
> So, what about we enable that feature if an Avocado-level configuration such as `vt.short_names_when_config` is set to `True` (having a default of `False`).

Thank you @clebergnu , I added the option `--vt-short-names-when-config` according to your comment.

clebergnu reviewed (approved):
LGTM, thanks!

luckyh commented on avocado_vt/loader.py:
```diff
@@ -199,7 +199,9 @@ def discover(self, url, which_tests=loader.DiscoverMode.DEFAULT):
             # Evaluate the proper avocado-vt test name
             test_name = None
-            if get_opt(self.config, 'vt.type') == "spice":
+            if get_opt(self.config, 'vt.short_names_when_config'):
```
Better to add one more expression `get_opt(self.config, 'vt.config')` if the option works only if using `--vt-config`, otherwise, I suggest introducing the option without the suffix `-when-config`.

PaulYuuu replied:
This option is not very useful for usage without config file since users can customize guests, machine_types, etc. in `avocado run` command. I will add `vt.config` together.

luckyh reviewed (approved):
LGTM

PaulYuuu commented:
Without `--vt-short-names-when-config`:
```
avocado run --journal --vt-config /var/lib/avocado/data/avocado-vt/backends/qemu/cfg/tests-example.cfg --dry-run
JOB ID     : 0000000000000000000000000000000000000000
JOB LOG    : /tmp/avocado-dry-run-xd8d6xkx/job-2020-12-15T21.21-0000000/job.log
 (001/197) io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (002/197) type_specific.io-github-autotest-qemu.rh_kernel_update: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (003/197) io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (004/197) type_specific.io-github-autotest-qemu.rh_kernel_update: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (005/197) io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (006/197) io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (007/197) io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (008/197) type_specific.io-github-autotest-qemu.rh_kernel_update: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (009/197) io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (010/197) type_specific.io-github-autotest-qemu.rh_kernel_update: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (011/197) io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (012/197) io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
```

With `--vt-short-names-when-config`:
```
avocado run --journal --vt-config /var/lib/avocado/data/avocado-vt/backends/qemu/cfg/tests-example.cfg --dry-run --vt-short-names-when-config
JOB ID     : 0000000000000000000000000000000000000000
JOB LOG    : /tmp/avocado-dry-run-04r7oty7/job-2020-12-15T21.26-0000000/job.log
 (001/197) power8_compat.Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.ppc64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (002/197) power8_compat.Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.ppc64.io-github-autotest-qemu.rh_kernel_update: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (003/197) power8_compat.Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (004/197) power8_compat.Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.ppc64le.io-github-autotest-qemu.rh_kernel_update: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (005/197) power8_compat.Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (006/197) Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (007/197) power8_compat.Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.9.ppc64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (008/197) power8_compat.Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.9.ppc64.io-github-autotest-qemu.rh_kernel_update: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (009/197) power8_compat.Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.9.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (010/197) power8_compat.Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.9.ppc64le.io-github-autotest-qemu.rh_kernel_update: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (011/197) power8_compat.Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (012/197) Host_RHEL.m8.u4.v0.product_av.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: CANCEL: Test cancelled due to --dry-run (0.00 s)
```

luckyh commented:
Thanks all.
